tests/sentry/incidents/action_handlers/test_slack.py

```diff
@@ -326,3 +328,92 @@ def test_alert_sent_recorded(self, mock_api_call, mock_record):
+        in a thread, and that critical alerts broadcast to the channel.
+        """
+        # Mocking Slack responses
+        initial_ts = "12345.67890"
+        update_ts = "98765.43210"
+        mock_post.side_effect = [
+            SlackResponse(
+                client=None,
+                http_verb="POST",
+                api_url="...",
+                req_args={},
+                data={"ok": True, "ts": initial_ts, "channel": self.channel_id},
+                headers={},
+                status_code=200,
+            ),
+            SlackResponse(
+                client=None,
+                http_verb="POST",
+                api_url="...",
+                req_args={},
+                data={"ok": True, "ts": update_ts, "channel": self.channel_id},
+                headers={},
+                status_code=200,
+            ),
+        ]
+        mock_api_call.return_value = {
+            "body": orjson.dumps({"ok": True}).decode(),
+            "headers": {},
+            "status": 200,
+        }
+
+        # 1. Fire initial warning alert
+        incident = self.create_incident(
+            alert_rule=self.alert_rule, status=IncidentStatus.WARNING.value
+        )
+        metric_value = 1000
+        with self.tasks():
+            self.handler.fire(
+                action=self.action,
+                incident=incident,
+                project=self.project,
+                metric_value=metric_value,
+                new_status=IncidentStatus.WARNING,
+            )
+
+        # Assert initial post
+        assert mock_post.call_count == 1
+        first_call_args, first_call_kwargs = mock_post.call_args
+        assert first_call_kwargs.get("thread_ts") is None
+        assert first_call_kwargs.get("reply_broadcast") is False
+        assert NotificationMessage.objects.filter(incident=incident).count() == 1
+        parent_message = NotificationMessage.objects.get(incident=incident)
+        assert parent_message.message_identifier == initial_ts
+        assert parent_message.parent_notification_message_id is None
+
+        # 2. Update incident status to critical and fire again
+        update_incident_status(
+            incident, IncidentStatus.CRITICAL, status_method=IncidentStatusMethod.MANUAL
+        )
+        metric_value_update = 2000
+        with self.tasks():
+            self.handler.fire(
+                action=self.action,
+                incident=incident,
+                project=self.project,
+                metric_value=metric_value_update,
+                new_status=IncidentStatus.CRITICAL,
+            )
+
+        # Assert update post is in thread and broadcast
+        assert mock_post.call_count == 2
+        second_call_args, second_call_kwargs = mock_post.call_args
+        assert second_call_kwargs.get("thread_ts") == initial_ts
+        assert second_call_kwargs.get("reply_broadcast") is True
+
+        # Assert new notification message is saved correctly
```

Comment:
nit: might be nice to break these two assertion groups into 2 tests; but seems like there's a lot of setup code 😅